src/Interpreters/ColumnAliasesVisitor.cpp

```diff
@@ -25,35 +25,15 @@ bool ColumnAliasesMatcher::needChildVisit(const ASTPtr & node, const ASTPtr &)
 
     return !(node->as<ASTTableExpression>()
             || node->as<ASTSubquery>()
-            || node->as<ASTArrayJoin>()
-            || node->as<ASTSelectQuery>()
-            || node->as<ASTSelectWithUnionQuery>());
+            || node->as<ASTArrayJoin>());
 }
 
 void ColumnAliasesMatcher::visit(ASTPtr & ast, Data & data)
 {
-    // If it's select query, only replace filters.
-    if (auto * query = ast->as<ASTSelectQuery>())
-    {
-        if (query->where())
-             Visitor(data).visit(query->refWhere());
-        if (query->prewhere())
-             Visitor(data).visit(query->refPrewhere());
```

Comment:
Now we don't visit select or select with union.
Tests are fine, but I wonder why did we need to visit filters before ...

Reply:
The original PR fails to make it work with array joins, thus it only rewrites PREWHERE/WHERE clauses.

Reply:
ok